src/ariel-os-stm32/src/lib.rs

```diff
@@ -78,6 +78,29 @@ pub fn init() -> OptionalPeripherals {
 
 // TODO: find better place for this
 fn board_config(config: &mut Config) {
+    #[cfg(any(context = "stm32f401ceux", context = "stm32f401cdux"))]
+    {
+        use embassy_stm32::rcc::*;
+
+        config.rcc.hse = Some(Hse {
+            freq: embassy_stm32::time::Hertz(25_000_000),
+            mode: HseMode::Oscillator,
+        });
+        config.rcc.pll_src = PllSource::HSE;
+        config.rcc.pll = Some(Pll {
+            prediv: PllPreDiv::DIV25,  // 1Hz
+            mul: PllMul::MUL336,       // 336Hz
+            divp: Some(PllPDiv::DIV4), // sysclk 84MHz
+            divq: Some(PllQDiv::DIV6), // USB 48MHz
```

Comment:
It might also be possible to reduce the pre-divider value, which is usually better as this reduces jitter and helps the PLL lock faster (we also do have plans for generating these clock configurations automatically later):
```suggestion
            prediv: PllPreDiv::DIV25,  // 1MHz
            mul: PllMul::MUL168,       // 168MHz
            divp: Some(PllPDiv::DIV2), // sysclk 84MHz
            divq: Some(PllQDiv::DIV3), // USB 48MHz
```
(There may be better clock configurations, not claiming this is optimal either.)